src/Entity/SnapshotManager.php

```diff
@@ -231,60 +233,40 @@ public function cleanup(PageInterface $page, $keep)
             throw new \RuntimeException(sprintf('Please provide an integer value, %s given', \gettype($keep)));
         }
 
-        $tableName = $this->getTableName();
-        $platform = $this->getConnection()->getDatabasePlatform()->getName();
-
-        if ('mysql' === $platform) {
-            return $this->getConnection()->exec(sprintf(
-                'DELETE FROM %s
-                WHERE
-                    page_id = %d
-                    AND id NOT IN (
-                        SELECT id
-                        FROM (
-                            SELECT id, publication_date_end
-                            FROM %s
-                            WHERE
-                                page_id = %d
-                            ORDER BY
-                                publication_date_end IS NULL DESC,
-                                publication_date_end DESC
-                            LIMIT %d
-                        ) AS table_alias
-                )',
-                $tableName,
-                $page->getId(),
-                $tableName,
-                $page->getId(),
-                $keep
-            ));
+        $innerQb = $this->getRepository()->createQueryBuilder('i');
+        $expr = $innerQb->expr();
+
+        // try a better Function expression for this?
+        $ifNullExpr = sprintf(
+            'CASE WHEN %s THEN 1 ELSE 0 END',
+            $expr->isNull('i.publicationDateEnd')
+        );
+
+        // Subquery DQL doesn't support Limit
+        $innerQb
+            ->select('i.id')
+            ->where($expr->eq('i.page', $page->getId()))
+            ->orderBy($ifNullExpr, Criteria::DESC)
+            ->addOrderBy('i.publicationDateEnd', Criteria::DESC)
+            ->setMaxResults($keep);
+
+        $query = $innerQb->getQuery();
+        if (method_exists($query, 'getSingleColumnResult')) {
```

Comment:
Hey @Hanmac why do you need this?

Reply:
It was introduced in 2.10.x. 

@jordisala1991 WDYT ? Should we add a stricter conflict about the version ? 
Or always using the else branch to avoid extra if/else ?

Reply:
i tried to raise the dependencies as low as possible, but i will follow your decisions

Reply:
same with this one:

```
        // for older doctrine dbal
        if (method_exists(Connection::class, 'executeStatement')) {
            $method = 'executeStatement';
        } else {
            $method = 'executeUpdate';
        }
```

Reply:
I think bumping the minimum minor version is totally fine

Reply:
doctrine bundle != doctrine orm != doctrine dbal version

i need to see which one i need to raise more

Reply:
and in theory it should be fine, it's just a minor version without BC breaks.

Reply:
> doctrine bundle != doctrine orm != doctrine dbal version
> 
> i need to see which one i need to raise more

yep, I misunderstood between `doctrine bundle` and `doctrine orm` 😅

Reply:
i bumped doctrine dependencies